web_src/js/components/RepoActionView.vue

```diff
@@ -335,6 +387,44 @@ const sfc = {
 
     isDone(status) {
       return ['success', 'skipped', 'failure', 'cancelled'].includes(status);
+    },
+
+    closeDropdown() {
+      if (this.menuVisible) this.menuVisible = false;
+    },
+
+    toggleTimeDisplay(type) {
+      const otherType = 'durationstamp'.replace(type, '');
+      this.timeVisible[type] = !this.timeVisible[type];
+      for (const el of document.querySelectorAll(`.log-time-${type}`)) {
+        toggleElem(el, this.timeVisible[type]);
+      }
+      // show at most one of log duration and timestamp (can be both invisible)
+      if (this.timeVisible[type] && this.timeVisible[otherType]) {
+        this.timeVisible[otherType] = false;
+        for (const el of document.querySelectorAll(`.log-time-${otherType}`)) {
```

Comment:
These selectors seem to broad. The scope should be limited into this component.

Reply:
`otherType` can only be 'duration' or 'stamp', and `.log-time-duration` and `.log-time-stamp` are classes only used in this component. Or do I misunderstood this?

Reply:
I mean "document.querySelectorAll" seems too broad. Could it only query the elements in this component?

Reply:
`document.querySelectorAll` here only accept `.log-time-duration` and `.log-time-stamp`, which are elements only occur in this component. Maybe explicitly check the type? https://github.com/go-gitea/gitea/pull/24876/commits/5e662223bdb5a7d765fce4e6f762f745c987539c

Reply:
I am not a fan of using `document.querySelectorAll` for a component's elements.

Here it isn't a problem, but it's not a best practice.

Think about a case, if there is a component, used multiple times on one page, the  `document.querySelectorAll`  would conflict others' states.


Generally speaking, to be strict, I think a component should only query elements belonging to itself.

Reply:
I see your point. It should use `ref` in Vue component, like discussed in https://github.com/go-gitea/gitea/pull/24876#issuecomment-1562104838 here the codes for generating log lines are native codes, we should find a way to turn these codes into a Vue component as well.

Reply:
>  we should find a way to turn these codes into a Vue component as well.

Nope, don't do this, otherwise you lose the full control of these elements.

These lines are written as it for performance purpose.

Reply:
I have left some comments there

<details>

![image](https://github.com/go-gitea/gitea/assets/2114189/fa833b05-750b-46d2-8f3e-ba58fb034ef2)

</details>


But I guess it needs more comments about why we want to construct these elements manually.